javaslang/src/main/java/javaslang/collection/LinkedHashMap.java

```diff
@@ -181,39 +468,203 @@ private LinkedHashMap(Queue<Tuple2<K, V>> list, HashMap<K, V> map) {
         return ofEntries(Collections.fill(n, (Supplier<? extends Tuple2<K, V>>) s));
     }
 
+    private static <K, V> Tuple2<K, V> t(java.util.Map.Entry<? extends K, ? extends V> e) {
+        return Tuple.of(e.getKey(), e.getValue());
+    }
+
     /**
-     * Creates a LinkedHashMap of the given entries.
+     * Creates a {@code LinkedHashMap} of the given entries.
      *
-     * @param entries Map entries
-     * @param <K>     The key type
-     * @param <V>     The value type
+     * @param <K> The key type
+     * @param <V> The value type
+     * @param e1  The 1st entry
      * @return A new Map containing the given entries
      */
     @SuppressWarnings("unchecked")
-    public static <K, V> LinkedHashMap<K, V> ofEntries(java.util.Map.Entry<? extends K, ? extends V>... entries) {
-        HashMap<K, V> map = HashMap.empty();
-        Queue<Tuple2<K, V>> list = Queue.empty();
-        for (java.util.Map.Entry<? extends K, ? extends V> entry : entries) {
-            final Tuple2<K, V> tuple = Tuple.of(entry.getKey(), entry.getValue());
-            map = map.put(tuple);
-            list = list.append(tuple);
-        }
-        return wrap(list, map);
+    public static <K, V> LinkedHashMap<K, V> ofEntries(java.util.Map.Entry<? extends K, ? extends V> e1) {
+        final HashMap<K, V> map = HashMap.ofEntries(e1);
+        final Queue<Tuple2<K, V>> list = Queue.of(t(e1));
+        return new LinkedHashMap<>(list, map);
     }
 
     /**
-     * Creates a LinkedHashMap of the given entries.
+     * Creates a {@code LinkedHashMap} of the given entries.
      *
-     * @param entries Map entries
-     * @param <K>     The key type
-     * @param <V>     The value type
+     * @param <K> The key type
+     * @param <V> The value type
+     * @param e1  The 1st entry
+     * @param e2  The 2nd entry
+     * @return A new Map containing the given entries
+     */
+    @SuppressWarnings("unchecked")
+    public static <K, V> LinkedHashMap<K, V> ofEntries(java.util.Map.Entry<? extends K, ? extends V> e1, java.util.Map.Entry<? extends K, ? extends V> e2) {
```

Comment:
this is not a vararg to avoid https://docs.oracle.com/javase/tutorial/java/generics/nonReifiableVarargsType.html#heap_pollution, right?
We do have many of these, though, maybe it's ok in our case :)